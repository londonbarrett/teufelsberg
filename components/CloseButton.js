import * as React from 'react';
import Router from 'next/router';

class CloseButton extends React.Component {
  onClick = (e) => {
    e.preventDefault();
    Router.push('/');
  }
  render() {
    return (
      <a
        className="link"
        href="/"
        onClick={this.onClick}
      >X
        <style jsx>
          {`
            .link {
              color: orange;
              display: block;
              height: 40px;
              right: 20px;
              overflow: hidden;
              position: absolute;
              text-indent: -9999px;
              top: 20px;
              width: 40px;
              transition: all .5s;
            }
            .link:hover {
              transform: rotate(180deg);
            }
            .link:before, .link:after {
              background: orange;
              content: '';
              height: 1px;
              left: 0;
              position: absolute;
              top: 50%;
              width: 100%;
            }
            .link:before {
              transform: rotate(45deg);
            }
            .link:after {
              transform: rotate(-45deg);
            }
          `}
        </style>
      </a>
    );
  }
}

export default CloseButton;
