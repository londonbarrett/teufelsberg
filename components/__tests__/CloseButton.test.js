/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import CloseButton from '../CloseButton';

describe('CloseButton Tests', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<CloseButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
