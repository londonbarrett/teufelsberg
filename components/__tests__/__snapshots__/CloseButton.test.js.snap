// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton Tests Renders correctly 1`] = `
<a
  className="jsx-348398248 link"
  href="/"
  onClick={[Function]}
>
  X
</a>
`;
